webpack/scenes/ContentViews/Details/Versions/Delete/RemoveSteps/CVVersionDeleteFinish.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useContext } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import { STATUS } from 'foremanReact/constants';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { selectRemoveCVVersionResponse, selectRemoveCVVersionStatus } from '../../../ContentViewDetailSelectors';
+import { getContentViewVersions, removeContentViewVersion } from '../../../ContentViewDetailActions';
+import Loading from '../../../../../../components/Loading';
+import DeleteContext from '../DeleteContext';
+
+const CVVersionDeleteFinish = () => {
+  const {
+    cvId, versionIdToRemove, versionEnvironments,
+    setIsOpen, selectedEnvSet,
+    selectedCVForAK, selectedEnvForAK, selectedCVForHosts,
+    selectedEnvForHost, affectedActivationKeys, affectedHosts, deleteFlow, removeDeletionFlow,
+  } = useContext(DeleteContext);
+  const removeCVVersionResponse = useSelector(state =>
+    selectRemoveCVVersionResponse(state, versionIdToRemove, versionEnvironments));
+  const removeCVVersionStatus = useSelector(state =>
+    selectRemoveCVVersionStatus(state, versionIdToRemove, versionEnvironments));
+  const removeResolved = removeCVVersionStatus === STATUS.RESOLVED;
+  const dispatch = useDispatch();
+  const [removeDispatched, setRemoveDispatched] = useState(false);
+  const selectedEnv = versionEnvironments.filter(env => selectedEnvSet.has(env.id));
+
+  useDeepCompareEffect(() => {
+    if (removeResolved && removeCVVersionResponse && removeDispatched) {
+      setIsOpen(false);
+      dispatch(getContentViewVersions(cvId));
+    }
+  }, [removeCVVersionResponse, removeResolved, setIsOpen, dispatch, cvId, removeDispatched]);
+
+  useDeepCompareEffect(() => {
+    if (!removeDispatched) {
+      const environmentIdParams = (deleteFlow || removeDeletionFlow) ?
```

Comment:
There's a lot going on here, for posterity can you add a few comments throughout this to explain the flow?

Reply:
We generally don't add comments. Although I have no issues with it, elders on team told me to avoid adding comments cause comments become a maintenance overhead soon.

Reply:
@Andrewgdewar @jeremylenz : Push me enough and I'll add comments 😄